server/asset/eth/eth.go

```diff
@@ -114,9 +108,8 @@ type Backend struct {
 	// method. signalMtx locks the blockChans array.
 	signalMtx  sync.RWMutex
 	blockChans map[chan *asset.BlockUpdate]struct{}
-	// The block cache stores just enough info about the blocks to prevent future
-	// calls to Block.
-	blockCache *blockCache
+	// The hash cache stores just enough info to detect reorgs.
```

Comment:
`... it also handles block polling and new block notifications to BlockChannel consumers.`

Reply:
Removed the blockCache in the last commit. Let me know how it looks.

Reply:
Looks like an improved design to me.  In `run` there is a polling loop, and if that's the only caller of `poll` and it's a synchronous call, then the fields of `best` don't need a mutex any more as it appears you've concluded as well.  And only needs to lock `blockChansMtx` if something actually changes, not each call to `poll`.

:+1: 

That the idea?

Reply:
Yes.